test-deployment/pom.xml

```diff
@@ -0,0 +1,150 @@
+            <groupId>jakarta.transaction</groupId>
+            <artifactId>jakarta.transaction-api</artifactId>
+        </dependency>
+        <!-- javax.transaction-api is test dependency -->
+        <dependency>
+            <groupId>org.jboss.spec.javax.transaction</groupId>
+            <artifactId>jboss-transaction-api_1.2_spec</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.marshalling</groupId>
+            <artifactId>jboss-marshalling</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jberet</groupId>
+            <artifactId>jberet-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jberet</groupId>
+            <artifactId>jberet-se</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.weld</groupId>
+            <artifactId>weld-core-impl</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.weld.se</groupId>
+            <artifactId>weld-se-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml</groupId>
+            <artifactId>aalto-xml</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.codehaus.woodstox</groupId>
+            <artifactId>stax2-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.openjdk.nashorn</groupId>
+            <artifactId>nashorn-core</artifactId>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>integration-test</goal>
+                            <goal>verify</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <argLine>${surefire.system.args}</argLine>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.wildfly.plugins</groupId>
+                <artifactId>wildfly-maven-plugin</artifactId>
+                <version>${version.org.wildfly.wildfly-maven-plugin}</version>
+                <configuration>
+                    <provisioning-dir>${jboss.home}</provisioning-dir>
+                    <offline>${galleon.offline}</offline>
+<!--                    <galleon-options>-->
+<!--                        <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>-->
```

Comment:
@jamezp It seems if I disabled this option then I can get the error from the `mvn wildfly:package` output:

```
ERROR: WFLYCTL0013: Operation ("add") failed - address: ([("extension" => "org.wildfly.extension.batch.jberet")])
java.lang.NoClassDefFoundError: org/jberet/spi/JobExecutor
        at org.wildfly.extension.batch.jberet@30.0.0.Final//org.wildfly.extension.batch.jberet._private.Capabilities.<clinit>(Capabilities.java:44)
        at org.wildfly.extension.batch.jberet@30.0.0.Final//org.wildfly.extension.batch.jberet.BatchSubsystemDefinition.<clinit>(BatchSubsystemDefinition.java:69)
        at org.wildfly.extension.batch.jberet@30.0.0.Final//org.wildfly.extension.batch.jberet.BatchSubsystemExtension.initialize(BatchSubsystemExtension.java:38)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.extension.ExtensionAddHandler.initializeExtension(ExtensionAddHandler.java:117)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.extension.ExtensionAddHandler.execute(ExtensionAddHandler.java:68)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:1048)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:769)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:460)
        at org.jboss.as.controller@22.0.1.Final//org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1433)

```

I guess the feature-pack is not loaded correctly.

Reply:
Yeah, it looks like the feature pack is not correct in some way. That said, I don't really know why you'd use the `package` goal. That provisions a WildFly server with a deployment.